### PR TITLE
Fix: libcrmcommon: handle schema versions properly

### PR DESCRIPTION
**Pull request by kgaillot**
As a byproduct, this eliminates a dependency on the math library.

### CODE REVIEW

File: lib/common/schemas.c [hunk 1]
```diff
@@ -134,19 +130,27 @@ get_schema_path(const char *name, const char *file)
     return crm_strdup_printf("%s/%s.rng", base, name);
 }
 
+static inline bool
+version_from_filename(const char *filename, int version[2])
+{
+    int rc = sscanf(filename, "pacemaker-%d.%d.rng", &version[0], &version[1]);
+
+    return (rc == 2)? TRUE : FALSE;
```
Comment (jnpkrn) [inline lib/common/schemas.c]: `return (rc == 2)` suffices

File: lib/common/schemas.c [hunk 2]
```diff
@@ -134,19 +130,27 @@ get_schema_path(const char *name, const char *file)
     return crm_strdup_printf("%s/%s.rng", base, name);
 }
 
+static inline bool
+version_from_filename(const char *filename, int version[2])
```
Comment (jnpkrn) [inline lib/common/schemas.c]: Repeated use of `int[2]` might perhaps justify a typedef'd type.
(put on the line where it became obvious to me)

File: lib/common/schemas.c [hunk 3]
```diff
@@ -164,29 +168,28 @@ schema_filter(const struct dirent *a)
...
-    /* crm_trace("%s (%f) vs. %s (%f) : %d", a[0]->d_name, a_version, b[0]->d_name, b_version, rc); */
-    return rc;
+    return 0;
 }
 
 static void
-__xml_schema_add(int type, float version, const char *name,
+__xml_schema_add(int type, int version[2], const char *name,
                  const char *location, const char *transform,
                  int after_transform)
```
Comment (jnpkrn) [inline lib/common/schemas.c]: `const int version[2]`?

**Comment by jnpkrn** [comment]
Do we need to reserve negative values for anything?  Otherwise I'd be in favor of `unsigned char` ("256 for a version part counter will be enough for everybody").

File: lib/common/schemas.c [hunk 4]
```diff
@@ -246,34 +255,35 @@ crm_schema_init(void)
     int lpc, max;
     const char *base = get_schema_root();
     struct dirent **namelist = NULL;
+    int zero[2] = {0, 0};
```
Comment (jnpkrn) [inline lib/common/schemas.c]: This could also be `const` then...

**Comment by jnpkrn** [comment]
Note that ceiling of 256 could also be good incentive not to bump the
versions excessively -- there are other connected costs (e.g. storage
for all the particular interim changes between point releases).

File: lib/common/schemas.c [hunk 5]
```diff
@@ -196,17 +199,23 @@ __xml_schema_add(int type, float version, const char *name,
...
+                                                         known_schemas[last].name);
     } else {
-        char dummy[1024];
         CRM_ASSERT(name);
         CRM_ASSERT(location);
-        sscanf(name, "%[^-]-%f", dummy, &version);
-        known_schemas[last].version = version;
+        sscanf(name, "%*[^-]-%d.%d",
+               &known_schemas[last].version[0],
+               &known_schemas[last].version[1]);
```
Comment (jnpkrn) [inline lib/common/schemas.c]: Also nice cleanup :)  (not that I am fluent with `sscanf, looked that up).

File: lib/common/schemas.c [hunk 6]
```diff
@@ -246,34 +255,35 @@ crm_schema_init(void)
...
-                sscanf(namelist[lpc+1]->d_name, "pacemaker-%f.rng",
-                       &next_version);
+                version_from_filename(namelist[lpc+1]->d_name, next_version);
 
-                if (floor(version) < floor(next_version)) {
+                if (version[0] < next_version[0]) {
                     struct stat s;
                     char *xslt = NULL;
 
-                    transform = crm_strdup_printf("upgrade-%.1f.xsl", version);
```
Comment (jnpkrn) [inline lib/common/schemas.c]: Apparently upgrading from `2.10` was unexpected in the old world order.

**Comment by jnpkrn** [comment]
On the more general note, following downside for this approach to
hitting the float-trackable versions stated in the past:

> float- and string-incompatible comparisons

which used to be noticable simplification especially in XSLT handling
(well, should be needed).  But we can live by and also XSLT should
not be impossible to handle (using substring cuts, etc.).

Looking at some other possibly affected software pieces:

* pcs looks fine:
  - https://github.com/ClusterLabs/pcs/blob/0.9.159/pcs/lib/cib/tools.py#L249
  - https://github.com/ClusterLabs/pcs/blob/0.9.159/pcs/utils.py#L99

* clufter:
  - FIXME: need to provide custom sorter at
    https://pagure.io/clufter/blob/a089855abffb64c81b49781e5785d6cfcdc69a71/f/format.py#_879

TL;DR: Not the end of the world :)

**Comment by kgaillot** [comment]
Added commits for review items and a little more cleanup
